src/util/copy/mod.rs

```diff
@@ -100,8 +100,16 @@ impl<VM: VMBinding> GCWorkerCopyContext<VM> {
     /// * `bytes`: The size of the object in bytes.
     /// * `semantics`: The copy semantic used for the copying.
     pub fn post_copy(&mut self, object: ObjectReference, bytes: usize, semantics: CopySemantics) {
-        // Clear forwarding bits.
-        object_forwarding::clear_forwarding_bits::<VM>(object);
```

Comment:
This is called for copying in both `CopySpace` and `ImmixSpace`. But the bulk zeroing is only implemented for `ImmixSpace` in this PR. With this change, we are not clearing side forwarding bits for `CopySpace`.

Also this happens when we copy an object (lazily clear forwarding bits). The change about bulk zeroing eagerly clears the forwarding bits in bulk. If it is meant to be an optimization, It is hard to say it is a clear win.

Reply:
I reverted the change of eagerly zeroing while sweeping.  I initially did it this way, similar to how I handled VO bits in the previous PR.  Then I realised there is a difference.  It is necessary to eagerly clear the VO bits for correctness, otherwise mutators may see the VO bit set for dead objects.  The forwarding bits are not observed by mutators, so it is not necessary to clear them during sweeping.

Now I clear the forwarding bits at the same time when clearing the mark bit.

Reply:
I also updated the comment to mention CopySpace, too, because both CopySpace and ImmixSpace clear the forwarding bits during preparation.